dlt/extract/incremental.py

```diff
@@ -180,9 +180,14 @@ def _wrap(*args: Any, **kwargs: Any) -> Any:
             elif isinstance(p.default, Incremental):
                 new_incremental = p.default.copy()
 
+            if new_incremental and not new_incremental.cursor_path:
+                # Workaround for optional arg in spec
+                new_incremental = None
+                bound_args.arguments[p.name] = None
+
             if not new_incremental:
```

Comment:
`Incremental` derives from `BaseConfiguration` which has this method
```
def is_partial(self) -> bool:
        """Returns True when any required resolvable field has its value missing."""
        if self.__is_resolved__:
            return False
        # check if all resolvable fields have value
        return any(
            field for field, hint in self.get_resolvable_fields().items() if getattr(self, field) is None and not is_optional_type(hint)
        )
        ```

I think you can just add
```
if not new_incremental or new_incremental.is_partial():
```

no need to set it to none above

Reply:
Very good!  changed to  https://github.com/dlt-hub/dlt/pull/244/files#diff-759977be85aa9d332107726916584d2574a442a8119117d9010bd5fa26be8310R183-R186